src/js/background/messageHandler.js

```diff
@@ -155,7 +155,24 @@ const messageHandler = {
           this.incrementCountOfContainerTabsOpened();
         }
 
-        backgroundLogic.unhideContainer(tab.cookieStoreId);
+        this.tabUpdateHandler = (tabId, changeInfo) => {
+          if (tabId === tab.id && changeInfo.status === "complete") {
+            // get current tab's url to not open the same one from hidden tabs
+            browser.tabs.get(tabId).then(loadedTab => {
+              backgroundLogic.unhideContainer(tab.cookieStoreId, loadedTab.url);
+            }).catch((e) => {
+              throw e;
+            });
+
+            browser.tabs.onUpdated.removeListener(this.tabUpdateHandler);
+          }
+        };
+
+        // if it's a container tab wait for it to complete and
+        // unhide other tabs from this container
+        if (tab.cookieStoreId.startsWith("firefox-container")) {
```

Comment:
Do we need this `if` check, or could we move this `addListener` into the above `if` that checks the `tab` is not `firefox-default` or `firefox-private` or a `moz-extension` url?

Reply:
yes, moving it makes perfect sense.  I'm not sure if 
`  firefox-default, firefox-private, nor our own confirm page loads `
would filter out all the non-container tabs, so added the additional check. Let me it you don't think it's necessary .